DataStructures/Lists/CircularBuffer.cs

```diff
@@ -0,0 +1,87 @@
+﻿using System;
+
+namespace DataStructures.Lists {
+    public class CircularBuffer<T> {
+        /// <summary>
+        /// Initializes a circular buffer with initial length of 10
+        /// </summary>
+        public CircularBuffer(bool canOverride=true) : this(10,canOverride) {
+        }
+
+        /// <summary>
+        /// Initializes a circular buffer with given length
+        /// </summary>
+        /// <param name="length">The length of the buffer</param>
+        public CircularBuffer(int length, bool canOverride=true) {
+            if (length < 1) {
+                throw new ArgumentOutOfRangeException("length can not be zero or negative");
```

Comment:
Maybe we should add test-case that cover that exception case?

Reply:
Yeah, I think so too.

Reply:
Added a test case that checked for length of 0 and -2. The test case was GREEN.